src/main/java/pl/connectis/spotifyapicli/authorization/TokenService.java

```diff
@@ -1,56 +1,34 @@
-package pl.connectis.Spotify_API_CLI.authorization;
+package pl.connectis.spotifyapicli.authorization;
 
-import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
-import lombok.EqualsAndHashCode;
-import lombok.Getter;
-import lombok.Setter;
-import lombok.ToString;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
 
 import java.io.*;
 
-@Getter
-@Setter
-@ToString
-@EqualsAndHashCode
-@JsonIgnoreProperties(ignoreUnknown = true)
 @Slf4j
-public class Token implements Serializable {
+@Component
+public class TokenService {
 
-    private final Long generation_time = System.currentTimeMillis();
     private final File codeFile = new File("token");
-    private String access_token;
-    private String token_type;
-    private String scope;
-    private Integer expires_in; //Seconds
-    private Long expirationTimeMillis;
-    private String refresh_token;
-
-    public Token() {
-    }
 
-    public void setExpirationTimeMillis() {
-        this.expirationTimeMillis = this.generation_time - 30000L + this.expires_in * 1000L; // 30 seconds overhead
-    }
+    public void saveTokenToFile(Token token) {
 
-    public Token saveToken() {
-        Token tokenToSave = this;
-        setExpirationTimeMillis();
+        token.setExpirationTimeInMillisecondsBasedOnGenerationTimeAndExpiresInSeconds();
 
         try {
             FileOutputStream fileOut = new FileOutputStream(codeFile);
             ObjectOutputStream out = new ObjectOutputStream(fileOut);
-            out.writeObject(tokenToSave);
+            out.writeObject(token);
             out.close();
             fileOut.close();
         } catch (IOException ex) {
             log.error("Saving file error " + ex.getMessage() + '\n' + "Path: " + codeFile.getPath());
         }
-        return tokenToSave;
     }
 
-    public Token readToken() {
-        Token tokenToRead = new Token();
+    //TODO don't like it I have to every time load Token from file if i want to access it, singleton or bean it anyway? but from bean i have to update values, maybe some lazy load bean?
+    public Token readTokenFromFile() {
```

Comment:
Wcześniej już na to zwracałem uwage. Dla mnie zbyt rozproszoną masz autoryzacje. Dzieje sie ona w paru miejscach a powinna w 1. `TokenService` powinien miec metode `getToken` i ona powinna być wołana w momencie wypełniania `HttpHeaderow`. 
Realizowała by ona całą logike pozyskiwania tokena.
coś w stylu:
```
getToken() {
      if(!isTokenValid(token)){
           token = authroizationStrategy.authorize();
           storeToken(token);
      }
      return token;
}
```
`token` to było by pole w klasie `TokenService`

metoda `readTokenFromFile` stała by sie prywatna. Ważne zeby zawsze zwracała null lub pusty token -`private static final EMPTY_TOKEN = new Token();` i ja zwracać w przypadku błędu.
Wołabym ją w konstruktorze tylko raz przy starcie apki
```
TokenService(){
   this.token = readTokenFromFile();
}
```
udało by sie - super - nie to nie